lib/spy.js

```diff
@@ -13,6 +13,165 @@ module.exports = function (chai, _) {
   var Assertion = chai.Assertion
     , flag = _.flag
     , i = _.inspect
+    , ID_KEY = typeof Symbol === 'undefined' ? '__id' : Symbol('id')
+    , STATE_KEY = typeof Symbol === 'undefined' ? '__state' : Symbol('state')
+    , spyAmount = 0
+    , DEFAULT_SANDBOX = new Sandbox()
+
+  /**
+   * # Sandbox constructor (function)
+   *
+   * Initialize new Sandbox instance
+   *
+   * @returns new sandbox
+   * @api private
+   */
+
+  function Sandbox() {
+    this[STATE_KEY] = {};
+  }
+
+  /**
+   * # Sandbox.on (function)
+   *
+   * Wraps an object method into spy assigned to sandbox. All calls will
+   * pass through to the original function.
+   *
+   *      var spy = chai.spy.sandbox();
+   *      var isArray = spy.on(Array, 'isArray');
+   *
+   * @param {Object} object
+   * @param {String} method name to spy on
+   * @returns created spy
+   * @api public
+   */
+
+  Sandbox.prototype.on = function (object, methodName) {
+    var method = chai.spy('object.' + methodName, object[methodName]);
+
+    method[ID_KEY] = ++spyAmount;
```

Comment:
Completely agree. If we do though, in the same PR we'll need something way to get the underlying calls, e.g. `chai.spy.getCall(spy, callNumber)`